src/cmd/wallet.rs

```diff
@@ -270,35 +272,73 @@ impl SpendOpt {
-                        );
+        let config = load_config(wallet_dir).context("loading config")?;
+
+        let output = match config.wallet_key {
+            WalletKey::SeedWordsFile {} => {
+                party.wallet().sign(&mut psbt, SignOptions::default())?;
+                let finalized = party
+                    .wallet()
+                    .finalize_psbt(&mut psbt, SignOptions::default())?;
+
+                let (output, txid) = cmd::decide_to_broadcast(
+                    party.wallet().network(),
+                    party.wallet().client(),
+                    psbt,
+                    yes,
+                    print_tx,
+                )?;
+
+                if let Some(txid) = txid {
+                    if !print_tx {
+                        for bet_id in claiming_bet_ids {
+                            if let Err(e) = party.take_next_action(bet_id, false) {
+                                eprintln!(
+                                    "error updating state of bet {} after broadcasting claim tx {}: {}",
+                                    bet_id, txid, e
+                                );
+                            }
+                        }
                     }
                 }
+                assert!(finalized, "transaction must be finalized at this point");
+                output
             }
-        }
-
+            WalletKey::Descriptor {
+                external: _,
+                internal: _,
+            } => {
+                let txid = psbt.clone().extract_tx().txid();
+
+                let mut psbt_file = PathBuf::from(config.psbt_output_dir.clone());
+                psbt_file.push(format!("{}.psbt", txid.to_string()));
+
+                fs::write(psbt_file.clone(), psbt.to_string())
+                    .context("writing PSBT file to SD path")?;
+                eprintln!("Wrote PSBT to {}", psbt_file.display());
+
+                let mut signed_psbt_file = PathBuf::from(config.psbt_output_dir.clone());
+                signed_psbt_file.push(format!("{}-signed.psbt", txid.to_string()));
+                eprintln!(
+                    "Please sign the PSBT and save it to {}. Press enter once signed.",
+                    signed_psbt_file.display()
+                );
+                let mut input = String::new();
+                let _ = std::io::stdin().read_line(&mut input);
+                let contents = fs::read_to_string(signed_psbt_file.clone())
+                    .with_context(|| format!("Reading PSBT file {}", signed_psbt_file.display()))?;
+                let psbt = PartiallySignedTransaction::from_str(&contents.trim())?;
+
+                let (output, _txid) = cmd::decide_to_broadcast(
+                    party.wallet().network(),
+                    party.wallet().client(),
+                    psbt,
+                    yes,
+                    print_tx,
+                )?;
+                output
+            }
+        };
```

Comment:
:bulb:  After looking at this I recalled BDK had a way of handling this: the [`Signer`] trait. What we should be doing is creating a SDPsbtSigner and implementing `Signer` for it and then adding it to our wallet. When asked to Sign it would prompt the user. This would mean we wouldn't need to know how we were signing anything in any of the application logic.

[`Signer`]: https://docs.rs/bdk/0.14.0/bdk/wallet/signer/trait.Signer.html

Reply:
Added `SDCardSigner`